akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/screenshare/ScreenshareRtmpBroadcastStartedVoiceConfEvtMsgHdlr.scala

```diff
@@ -61,8 +63,15 @@ trait ScreenshareRtmpBroadcastStartedVoiceConfEvtMsgHdlr {
         log.info("START broadcast ALLOWED when isBroadcastingRTMP=false")
 
         ScreenshareDAO.insert(liveMeeting.props.meetingProp.intId, liveMeeting.screenshareModel)
-        Layouts.setScreenshareAsContent(liveMeeting.layouts, true)
-        LayoutDAO.insertOrUpdate(liveMeeting.props.meetingProp.intId, liveMeeting.layouts)
+        if (!Layouts.getScreenshareAsContent(liveMeeting.layouts)) {
+          Layouts.setScreenshareAsContent(liveMeeting.layouts, true)
+          LayoutDAO.insertOrUpdate(liveMeeting.props.meetingProp.intId, liveMeeting.layouts)
+          for {
+            presenter <- findPresenter(liveMeeting.users2x)
+          } yield {
+            ScreenshareAsContenthdlrHelper.sendSetScreenshareAsContentEvtMsg(presenter.intId, liveMeeting, bus.outGW)

```

Comment:
Just to raise the discussion: is setting the screenshare as a content necessary in this case? To my knowledge, the message `SetScreenshareAsContentEvtMsg` will only take care of the layout in the recording (I mean the swap transition event that need to happen in the recording). While the event `StartWebRTCDesktopShareEvent` is generate by `webRTC-SFU`  and already deals with showing it in the recordings. So, correct me if I'm wrong, but this line, to me, seems like a duplication of behaviour (showing the screenshare in the recordings).

Reply:
Ok, got it, thank you!